backend/file-system/src/main/java/com/htec/filesystem/service/FileService.java

```diff
@@ -204,45 +205,89 @@ public void updateDeletedFiles(AuthUser user, List<Long> fileIds, Boolean delete
 
     void recover(AuthUser user, List<FileEntity> fileEntities) {
 
+        for (FileEntity fileEntity : fileEntities) {
+            Long parentFolderId = fileEntity.getParentFolderId();
+            Optional<FolderEntity> folderEntity = folderRepository.findById(parentFolderId);
```

Comment:
do not fetch items from db within a list? You can prepare ids and use findByIdIn()